conformance/tests/httproute-cors.go

```diff
@@ -0,0 +1,280 @@
+						"access-control-request-headers": "x-header-1, x-header-2",
+					},
+				},
+				// Set the expected request properties and namespace to empty strings.
+				// This is a workaround to avoid the test failure.
+				// The response body is empty because the request is a preflight request,
+				// so we can't get the request properties from the echoserver.
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Host:    "",
+						Method:  "OPTIONS",
+						Path:    "",
+						Headers: nil,
+					},
+				},
+				Namespace: "",
+				Response: http.Response{
+					StatusCode: 200,
+					Headers: map[string]string{
+						"access-control-allow-origin":      "https://www.bar.com",
+						"access-control-allow-methods":     "GET, OPTIONS",
+						"access-control-allow-headers":     "x-header-1, x-header-2",
+						"access-control-expose-headers":    "x-header-3, x-header-4",
+						"access-control-max-age":           "3600",
+						"access-control-allow-credentials": "true",
+					},
+				},
+			},
+			{
+				TestCaseName: "CORS preflight request from a non-matching origin should not be allowed",
+				Request: http.Request{
+					Path:   "/cors-1",
+					Method: "OPTIONS",
+					Headers: map[string]string{
+						"Origin":                        "https://foobar.com",
+						"access-control-request-method": "GET",
+					},
+				},
+				// Set the expected request properties and namespace to empty strings.
+				// This is a workaround to avoid the test failure.
+				// The response body is empty because the request is a preflight request,
+				// so we can't get the request properties from the echoserver.
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Host:    "",
+						Method:  "OPTIONS",
+						Path:    "",
+						Headers: nil,
+					},
+				},
+				Namespace: "",
+				Response: http.Response{
+					AbsentHeaders: []string{
+						"access-control-allow-origin",
+					},
+				},
+			},
+			{
+				TestCaseName: "Simple request from an exact mactching origin should be allowed",
+				Namespace:    ns,
+				Request: http.Request{
+					Path:   "/cors-1",
+					Method: "GET",
+					Headers: map[string]string{
+						"Origin":                         "https://www.foo.com",
+						"access-control-request-method":  "GET",
+						"access-control-request-headers": "x-header-1, x-header-2",
+					},
+				},
+				Response: http.Response{
+					StatusCode: 200,
+					Headers: map[string]string{
+						"access-control-allow-origin": "https://www.foo.com",
+					},
+				},
+			},
+			{
+				TestCaseName: "Simple request from a wildcard matching origin should be allowed",
+				Namespace:    ns,
+				Request: http.Request{
+					Path:   "/cors-1",
+					Method: "GET",
+					Headers: map[string]string{
+						"Origin":                         "https://www.bar.com",
+						"access-control-request-method":  "GET",
+						"access-control-request-headers": "x-header-1, x-header-2",
```

Comment:
According to the specification (https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/httproute_types.go#L1386),  headers (and also methods) are separated by `","` instead of `", "`.

Reply:
I've tested this against Envoy Gateway, and the `access-control-request-headers` headers include ", ". Also, the [example](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Methods#examples) in https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Methods contains whitespaces.

```
Access-Control-Allow-Methods: PUT, DELETE
```

Reply:
Maybe we are both right and the white space is optional: https://httpwg.org/specs/rfc9110.html#rfc.section.5.3

Reply:
Yeah, you're right. The test should accommodate both formats.

Reply:
@snorwin I added an option to ignore the whitespace in the header value.